pkg/github/github_secrets.go

```diff
@@ -0,0 +1,127 @@
+package github
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"net/http"
+	"net/url"
+
+	"github.com/google/go-github/v53/github"
+	"golang.org/x/crypto/nacl/box"
+	"golang.org/x/oauth2"
+)
+
+func encodeWithPublicKey(text string, publicKey string) (string, error) {
+	publicKeyBytes, err := base64.StdEncoding.DecodeString(publicKey)
+	if err != nil {
+		return "", err
+	}
+
+	var publicKeyDecoded [32]byte
+	copy(publicKeyDecoded[:], publicKeyBytes)
+
+	encrypted, err := box.SealAnonymous(nil, []byte(text), (*[32]byte)(publicKeyBytes), rand.Reader)
+	if err != nil {
+		return "", err
+	}
+
+	encryptedBase64 := base64.StdEncoding.EncodeToString(encrypted)
+
+	return encryptedBase64, nil
+}
+
+func createRepositorySecret(privateURL, owner, repo, env, secretName, secretValue, authToken string) error {
+	ctx := context.Background()
+	client, err := newGithubClientWithToken(authToken, privateURL)
+	if err != nil {
+		return err
+	}
+
+	var key *github.PublicKey
+	var repositoryID int
+
+	if env == "" {
+		key, _, err = client.Actions.GetRepoPublicKey(ctx, owner, repo)
+		if err != nil {
+			return err
+		}
+	} else {
+		repositoryID, err = getRepoID(ctx, client, owner, repo)
+		if err != nil {
+			return err
+		}
+		key, _, err = client.Actions.GetEnvPublicKey(ctx, repositoryID, env)
+		if err != nil {
+			return err
+		}
+	}
+
+	encryptedSecret, err := encodeWithPublicKey(secretValue, *key.Key)
+	if err != nil {
+		return err
+	}
+
+	secret := &github.EncryptedSecret{
+		Name:           secretName,
+		EncryptedValue: encryptedSecret,
+	}
+
+	if env == "" {
+		_, err := client.Actions.CreateOrUpdateRepoSecret(ctx, owner, repo, secret)
+		return err
+	}
+
+	_, err = client.Actions.CreateOrUpdateEnvSecret(ctx, repositoryID, env, secret)
```

Comment:
Suggest to split up this function, one to create repo secret, and one to create an env secret. It's not that clear that the `env` parameter can be empty, especially when no other parameter can.

Reply:
done in [699d745](https://github.com/jadolg/klum/pull/19/commits/699d745e7ea069f3986e1a2a5f046b154803fab3)